dipy/align/metrics.py

```diff
@@ -244,6 +244,21 @@ def initialize_iteration(self):
         re-orienting the gradients in the voxel space using the corresponding
         affine transformations.
         """
+        min_size = self.radius * 2 + 1
+
+        def invalid_image_size(image):
```

Comment:
I would simplify this function using something like:
e.g.
```Python
def invalid_image_size(image):
    return any([size < min_size for size in image.shape])
```
Given that this is now only a single line, it may not be worth making a separate function for it, but that is just a stylistic choice.

Reply:
wow, Nice! I should definitively use more `any` and `all`